### ISSUE TITLE
Fails on template-curly-spacing rule in JSX code

### ISSUE THREAD

**Issue by pistonsky** [eslint/eslint#14876]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `eslint@7.32.0` for the project I'm working on.

The issue I was facing is with templated strings like these:
```js
<Text style={styles.topText}>{`${topText.value === null ? '' : topText.value}${topText.suffix}`}</Text>
```

When I tried to lint my code, eslint failed, showing me the line where problem occured. No surprise, because `token` is allowed to be null:
```js
    /**
     * Gets the first token of the given node.
     * @param {ASTNode} node The AST node.
     * @param {number|Function|Object} [options=0] The option object. If this is a number then it's `options.skip`. If this is a function then it's `options.filter`.
     * @param {boolean} [options.includeComments=false] The flag to iterate comments as well.
     * @param {Function|null} [options.filter=null] The predicate function to choose tokens.
     * @param {number} [options.skip=0] The count of tokens the cursor skips.
     * @returns {Token|null} An object representing the token.
     */
    getFirstToken(node, options) {
        return createCursorWithSkip(
            cursors.forward,
            this[TOKENS],
            this[COMMENTS],
            this[INDEX_MAP],
            node.range[0],
            node.range[1],
            options
        ).getOneToken();
    }
```

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/eslint/lib/rules/template-curly-spacing.js b/node_modules/eslint/lib/rules/template-curly-spacing.js
index 26043bc..9c174ab 100644
--- a/node_modules/eslint/lib/rules/template-curly-spacing.js
+++ b/node_modules/eslint/lib/rules/template-curly-spacing.js
@@ -49,7 +49,7 @@ module.exports = {
          * @returns {void}
          */
         function checkSpacingBefore(token) {
-            if (!token.value.startsWith("}")) {
+            if (!token?.value.startsWith("}")) {
                 return; // starts with a backtick, this is the first template element in the template literal
             }
 
@@ -92,7 +92,7 @@ module.exports = {
          * @returns {void}
          */
         function checkSpacingAfter(token) {
-            if (!token.value.endsWith("${")) {
+            if (!token?.value.endsWith("${")) {
                 return; // ends with a backtick, this is the last template element in the template literal
             }
 
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by nzakas** [comment]
Please update your issue to use our [bug report template](https://github.com/eslint/eslint/blob/master/.github/ISSUE_TEMPLATE/BUG_REPORT.md) and we will take a look. Thanks.

**Comment by nzakas** [comment]
No response, so closing.
